### PR TITLE
Organization setting has been stored in the database

### PR DESCRIPTION
**Pull request by VivekU03**
#2122 

### Description
Created a new setting with the name organization and created a table in database and and stored organization setting details in the database.

- [x] I have performed a self-review of my own code.

### CODE REVIEW

File: app/Http/Controllers/Settings/SettingController.php [hunk 1]
```diff
@@ -39,4 +41,25 @@ public function updateInvoiceTemplates(SettingRequest $request)
...
+            'name' => $request->name,
+            'address' => $request->address,
+            'annual_sales' => $request->annual_sales,
+            'members' => $request->members,
+            'industry' => $request->industry,
+            'email' => $request->email,
+            'billing_details' => $request->billing_details,
+            'website' => $request->website,
+        ]);
+
```
Comment (gautam071) [inline app/Http/Controllers/Settings/SettingController.php:62]: Can you add the backend validation

File: resources/views/settings/organization/organization.blade.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        </div>
+        <div class="ml-2 mt-1">
+        <button type="submit" class="btn btn-primary" id="save-btn-action">Submit</button>
+        </div>
+        </form>
+    </div>
+</div>
+
+
+@stop
```
Comment (gautam071) [inline resources/views/settings/organization/organization.blade.php:56]: `@stop` is not recommended to use. 
Can we change it to the @endsection

If you want to know more https://stackoverflow.com/questions/21199412/laravel-blade-endsection-vs-stop

File: resources/views/settings/index.blade.php [hunk 3]
```diff
@@ -45,6 +45,17 @@
...
+        
+        @can('organization_settings.view')
+        <div class="col-md-4">
+            <div class="card h-75 mx-4 mt-3 mb-5 ">
+                <a class="card-body no-transition" href="{{ route('setting.organization') }}">
+                    <br><h2 class="text-center">Organization</h2><br>
+                </a>
+            </div>
+        </div>
+        @endcanany
```
Comment (gautam071) [inline resources/views/settings/index.blade.php:57]: You started the `@can` and using `@endcanany`?
What was the reason?

File: resources/views/settings/organization/organization.blade.php [hunk 4]
```diff
@@ -0,0 +1,56 @@
+@extends('layouts.app')
+
+@section('content')
+<div class="card-body">
+    <div>
+        <form action="{{route('setting.create.organization')}}" method="POST">
+            @csrf
+        <div class="form-group col-md-5">
+            <label for="name" class="field-required">Name</label>
+            <input type="text" class="form-control" name="name" placeholder="Organization"
```
Comment (gautam071) [inline resources/views/settings/organization/organization.blade.php:10]:
```suggestion
            <input type="text" class="form-control" name="name" placeholder="Organization Name"
```

File: resources/views/settings/organization/organization.blade.php [hunk 5]
```diff
@@ -0,0 +1,56 @@
+@extends('layouts.app')
+
+@section('content')
+<div class="card-body">
+    <div>
+        <form action="{{route('setting.create.organization')}}" method="POST">
+            @csrf
+        <div class="form-group col-md-5">
+            <label for="name" class="field-required">Name</label>
```
Comment (gautam071) [inline resources/views/settings/organization/organization.blade.php:9]:
```suggestion
            <label for="name" class="field-required">Organization Name</label>
```

**Review by gautam071 — CHANGES_REQUESTED** [review]
Few comments

File: app/Http/Controllers/Settings/SettingController.php [hunk 6]
```diff
@@ -39,4 +41,26 @@ public function updateInvoiceTemplates(SettingRequest $request)
...
+            'name' => $validatedData['name'],
+            'address' => $validatedData['address'],
+            'annual_sales' => $validatedData['annual_sales'],
+            'members' => $validatedData['members'],
+            'industry' => $validatedData['industry'],
+            'email' => $validatedData['email'],
+            'billing_details' => $validatedData['billing_details'],
+            'website' => $validatedData['website'],
+        ]);
+
```
Comment (gautam071) [inline app/Http/Controllers/Settings/SettingController.php:63]: Can we use a service file to store the data

File: app/Http/Requests/OrganizationRequest.php [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+            ],
+            'billing_details' => [
+                'required',
+                'string'
+            ],
+            'website' => [
+                'required',
+                'string'
+            ],
+        ];
```
Comment (gautam071) [inline app/Http/Requests/OrganizationRequest.php:59]: Can you check the format of the rules

File: app/Models/Organization.php [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Organization extends Model
+{
+    use HasFactory;
+    protected $table = 'organizations';
```
Comment (gautam071) [inline app/Models/Organization.php:11]:
```suggestion

    protected $table = 'organizations';
```

File: resources/views/settings/index.blade.php [hunk 9]
```diff
@@ -45,6 +45,17 @@
...
+        @can('organization_settings.view')
+        <div class="col-md-4">
+            <div class="card h-75 mx-4 mt-3 mb-5 ">
+                <a class="card-body no-transition" href="{{ route('setting.organization') }}">
+                    <br><h2 class="text-center">Organization</h2><br>
+                </a>
+            </div>
+        </div>
+        @endcan
+
```
Comment (gautam071) [inline resources/views/settings/index.blade.php:58]: empty space

File: resources/views/settings/organization/organization.blade.php [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+@section('content')
+<div class="card-body">
+    <div>
+        @if(session('message'))
+        <div class="alert alert-success">
+            <h4>{{ session('message') }}
+        </div>
+        @endif
+        <form action="{{route('setting.create.organization')}}" method="POST">
+            @csrf
```
Comment (gautam071) [inline resources/views/settings/organization/organization.blade.php:12]: Indentation

File: resources/views/settings/organization/organization.blade.php [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+        <div class="form-group col-md-5">
+            <label for="name" class="field-required">Name</label>
+            <input type="text" class="form-control" name="name" placeholder="Organization" required>
+                @error('name') <small class="text-danger">{{ $message }}</small>@enderror
```
Comment (gautam071) [inline resources/views/settings/organization/organization.blade.php:16]: format

File: resources/views/settings/organization/organization.blade.php [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+        </div>
+        <div class="form-group col-md-5">
+            <label for="address" class="field-required">Address</label>
+            <input type="text" class="form-control" name="address" placeholder="Address" required>
+            @error('address') <small class="text-danger">{{ $message }}</small>@enderror
```
Comment (gautam071) [inline resources/views/settings/organization/organization.blade.php:21]: format

File: resources/views/settings/organization/organization.blade.php [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+        </div>
+        <div class="form-group col-md-5">
+            <label for="annual_sale" class="field-required">Annual Sales</label>
+            <input type="number" class="form-control" name="annual_sales"  placeholder="Annual Sales" required>
+            @error('annual_sales') <small class="text-danger">{{ $message }}</small>@enderror
```
Comment (gautam071) [inline resources/views/settings/organization/organization.blade.php:26]: Check the format in the blade file

File: resources/views/settings/organization/organization.blade.php [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+        </div>
+        <div class="ml-2 mt-1">
+        <button type="submit" class="btn btn-primary" id="save-btn-action">Submit</button>
+        </div>
+        </form>
+    </div>
+</div>
+
+
+@endsection
```
Comment (gautam071) [inline resources/views/settings/organization/organization.blade.php:61]: line at end of file

**Review by gautam071 — CHANGES_REQUESTED** [review]
Left some comments

File: app/Http/Controllers/Settings/SettingController.php (hunk 6 above)
Comment (VivekU03) [reply app/Http/Controllers/Settings/SettingController.php:63]: Task required me to store the data in the database. Please confirm where to store the data.

File: app/Http/Controllers/Settings/SettingController.php [hunk 15]
```diff
@@ -39,4 +46,18 @@ public function updateInvoiceTemplates(SettingRequest $request)
 
         return redirect()->back()->with('status', 'Settings saved!');
     }
+
+    public function organizationData()
+    {
+        return view('settings.organization.organization');
+    }
```
Comment (gautam071) [inline app/Http/Controllers/Settings/SettingController.php:53]: What's the use of this?

**Review by gautam071 — CHANGES_REQUESTED** [review]
Looks good just a question

File: app/Http/Controllers/Settings/SettingController.php (hunk 15 above)
Comment (VivekU03) [reply app/Http/Controllers/Settings/SettingController.php:53]: To view this page.
Screenshot

.

Comment (VivekU03) [reply app/Http/Controllers/Settings/SettingController.php:53]: Sir please check it.

Comment (gautam071) [reply app/Http/Controllers/Settings/SettingController.php:53]: The index()?

**Review by gautam071 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**